### PR TITLE
Implementation of Maximin Support New Method using NetworkX (rule_id: maximin-support) - verified

### PR DESCRIPTION
**Pull request by AmitGini**
**Closes** #99

Hi,
I have implemented the new method to compute maximin support based on the paper *"Efficient, traceable, and numerical error-free"* by Luis Sánchez-Fernández (October 2023).

* **Paper:** [https://arxiv.org/abs/2309.15104](https://arxiv.org/abs/2309.15104)
* **Rule ID:** `maximin-support`
* **Algorithm:** `nx-max-flow`

**Implementation Details:**

* **New Module:** Added `abcrules_networkx.py` for this implementation and future graph-based rules using the existing `networkx` dependency.
* **Ordering:** I observed that implementations are generally ordered by execution speed/complexity, so I placed this function accordingly within the codebase.

**Testing & Validation:**
* **Marker:** Added a new marker: `@pytest.mark.networkx`.
* **Doctests:** Added examples to both functions, command to run: `python -m doctest abcvoting/abcrules_networkx.py -v`
* **Logging:** Added INFO, DEBUG, and WARNING loggers to the implementation. *(Note: I did not include the `example.py` profile script in this PR to keep the diff clean - [example.py](https://github.com/user-attachments/files/24361236/example.py)).*

I have also added specific edge-case tests to `tests_abcrules.py`:
1. `test_maximin_support_nx_max_flow_random_profiles`: Tests Variety sizes of Random profiles.
2. `test_maximin_support_nx_invalid_committeesize`: Ensures `committeesize <= 0` raises an error.
3. `test_maximin_support_nx_invalid_committeesize_exceeds_candidates`: Ensures `committeesize > num_cand` raises an error.
4. `test_maximin_support_nx_with_too_few_approved_candidates_and_max_num_of_committee`: Verifies correct behavior when approved candidates < committee size (with max_num constraints).
5. `test_maximin_support_nx_max_flow_irresolute_and_max_num_of_committee`: Verifies correct return size in irresolute mode.

**Questions & Request for Feedback**
I would appreciate your review of the entire implementation. Please let me know if there are any components—such as function documentation, doctests, pytests, or loggers—that need to be fixed, removed, or expanded.

I am particularly interested in your thoughts on the **random testing strategy**:

Since the output is non-deterministic (multiple correct winning committees are possible), comparing results directly is difficult. I utilized the existing `generate.random_profile` with an IC distribution (`random.uniform(0.1, 0.9)`).

To validate correctness, I compared my NetworkX output against the Gurobi implementation (with `resolute=False`). I checked if the committee returned by my implementation is **contained** within the set of committees returned by Gurobi.
*(Note: The Gurobi output is capped at 50 possible committees).*

* Is this the preferred way to verify non-deterministic outputs in this repository?
* Currently, I have **not** marked this test as `@pytest.mark.slow`. Should I add this marker?

**Results:**
All tests passed:
`6933 passed, 142065 deselected in 139.30s (0:02:19)`

### CODE REVIEW

**Comment by AmitGini** [comment]
Hi @martinlackner 

I finished what you asked for, let me know if anything else comes up.

Best regards,
Amit

File: abcvoting/abcrules_networkx.py [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+    -------
+    float
+        The maximin support value for the committee S.
+        Returns 0.0 if no support can be achieved or if S is empty.
+
+    Algorithm Overview
+    ------------------
+    Notions:
+    - S: Test committee (set of candidate indices)
+    - k_S_size = |S|: Size of the test committee
```
Comment (martinlackner) [inline abcvoting/abcrules_networkx.py]: k_S_size is the committee size, right? Since this is called committee_size throughout the repo, I would prefer to stick to this name

File: abcvoting/abcrules_networkx.py [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+        >>> scores[4]               # committee: [3,0,4]
+        1.0
+
+    ----------------------------------------------------------------
+        Example 2: Weighted voters
+        Testing that voter weights are properly considered
+
+        # Setup profile with weights
+        >>> profile = Profile(3)
+        >>> profile.add_voter(Voter([1, 2], weight=1))  # Weight 2
```
Comment (martinlackner) [inline abcvoting/abcrules_networkx.py]: comment is switched with the next line
